fissa/core.py

```diff
@@ -752,33 +826,40 @@ def separate(self, redo_prep=False, redo_sep=False):
         # Do the extraction
         if use_multiprocessing and sys.version_info < (3, 0):
             # define pool
-            pool = Pool(self.ncores_separation)
-
+            pool = multiprocessing.Pool(self.ncores_separation)
             # run separation
-            results = pool.map(separate_func, inputs)
+            outputs = pool.map(
+                _separate_wrapper,
+                zip(
+                    self.raw,
+                    range(n_roi),
+                    itertools.repeat(self.alpha, n_roi),
+                    itertools.repeat(self.method, n_roi),
+                ),
+            )
             pool.close()
             pool.join()
 
         elif use_multiprocessing:
-            with Pool(self.ncores_separation) as pool:
+            with multiprocessing.Pool(self.ncores_separation) as pool:
                 # run separation
-                results = pool.map(separate_func, inputs)
+                outputs = pool.starmap(_separate_cfg, zip(self.raw, range(n_roi)))
```

Comment:
So I was working on combining theses changes with tqdm and found that there's not an easy way to do starmap with tqdm: https://stackoverflow.com/questions/57354700/starmap-combined-with-tqdm

https://stackoverflow.com/questions/50154204/tqdm-progress-bar-and-multiprocessing

I'll use a normal imap I suppose.

Reply:
You can put tqdm around the arguments https://stackoverflow.com/a/65854996/
```suggestion
                outputs = pool.starmap(_separate_cfg, tqdm.tqdm(zip(self.raw, range(n_roi))))
```